experiment/simulation/js/fa.js

```diff
@@ -0,0 +1,333 @@
+import { registerGate } from "./main.js";
+import { setPosition } from "./layout.js";
+import {gates} from './gate.js';
+import {jsPlumbInstance} from "./main.js";
+
+// Dictionary of all full adders in the circuit with their IDs as keys
+let fullAdder = {}
+
+function clearFAs() {
+
+    for (let faID in fullAdder) {
+        delete fullAdder[faID];
+    }
+    fullAdder = {};
+}
+
+// {output-id: [gate,pos]}
+let finalOutputs = {
+    "Output-5": [],

```

Comment:
If you do not need hyphens here, you'll not need to put the keys in quotes

Reply:
But we do need hypens as it follows a particular format of the IDs of the components ( <Component type>-<Unique number>)